pkg/advisory/discover_aliases.go

```diff
@@ -0,0 +1,150 @@
+package advisory
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"sort"
+
+	"github.com/samber/lo"
+	"github.com/wolfi-dev/wolfictl/pkg/configs"
+	v2 "github.com/wolfi-dev/wolfictl/pkg/configs/advisory/v2"
+	"github.com/wolfi-dev/wolfictl/pkg/vuln"
+)
+
+// DiscoverAliasesOptions is the set of options for the DiscoverAliases
+// function.
+type DiscoverAliasesOptions struct {
+	// AdvisoryDocs is the Index of advisory documents on which to operate.
+	AdvisoryDocs *configs.Index[v2.Document]
+
+	// AliasFinder is the alias finder to use for discovering aliases for the given
+	// vulnerabilities.
+	AliasFinder AliasFinder
+
+	// SelectedPackages is the list of packages to operate on. If empty, all
+	// packages will be operated on.
+	SelectedPackages []string
+}
+
+// DiscoverAliases queries external data sources for aliases for the
+// vulnerabilities described in the selected advisories and updates the advisory
+// documents with the discovered aliases.
+func DiscoverAliases(ctx context.Context, opts DiscoverAliasesOptions) error {
+	documents := opts.AdvisoryDocs.Select().Configurations()
+
+	// If the caller selected specific packages (by name), filter the set of
+	// advisory documents to just those packages.
+	if len(opts.SelectedPackages) > 0 {
+		documents = lo.Filter(documents, func(doc v2.Document, _ int) bool {
+			return slices.Contains(opts.SelectedPackages, doc.Package.Name)
+		})
```

Comment:
We'd have an opportunity to make `opts.SelectedPackages` a set instead of a slice. Probably not a problem right now, but if we have many packages, that could be slow.

Reply:
Done!